src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php

```diff
@@ -88,16 +97,19 @@ public function parse($expression, $names)
         $cacheKeyItems = array();
 
         foreach ($names as $nameKey => $name) {
-            $cacheKeyItems[] = is_int($nameKey) ? $name : $nameKey.':'.$name;
+            $cacheKeyItems[] = is_int($nameKey) ? $name : $nameKey.'='.$name;
         }
 
-        $key = $expression.'//'.implode('|', $cacheKeyItems);
+        $key = $this->sanitizeCacheKey($expression.'//'.implode('|', $cacheKeyItems));
+
+        $cacheItem = $this->cache->getItem($key);
 
-        if (null === $parsedExpression = $this->cache->fetch($key)) {
+        if (null === $parsedExpression = $cacheItem->get()) {
```

Comment:
I agree with @ReDnAxE there is no need for isHit here and allows writting the code on a single line. Simpler to me.